### PR TITLE
[ADD] Field Service Change Management

### PR DESCRIPTION
**Pull request by patrickrwilson**
This adds a new module to the Field Service family that provides a way to create change logs on locations. Our need was to be able to track changes that happen at a location. These changes could be structural changes or configuration changes for example. There are reports where we can print historical changes at one or more locations.

### CODE REVIEW

**Comment by patrickrwilson** [comment]
Ping @max3903 @osi-scampbell

File: fieldservice_change_management/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        'data/change_log_type.xml',
+        'report/change_log_reports.xml',
+        'views/change_log.xml',
+        'views/change_log_impact.xml',
+        'views/change_log_stage.xml',
+        'views/change_log_tags.xml',
+        'views/change_log_type.xml',
+        'views/fsm_location.xml',
+    ],
+    'application': True,
```
Comment (max3903) [inline fieldservice_change_management/__manifest__.py:30]:
```suggestion
    'application': True,
    'development_status': 'Beta',
    'maintainers': ['patrickrwilson'],
```

File: fieldservice_change_management/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+{
+    'name': 'Field Service - Change Management',
+    'summary': 'Manage Change Logs on Locations',
+    'author': 'Patrick Wilson: Pavlov Media',
+    'website': 'https://www.pavlovmedia.com',
+    'category': 'Extra Tools',
+    'version': '12.0.1.0.0',
+    'license': 'Other proprietary',
+    'depends': ['mail',
+                'fieldservice',
```
Comment (max3903) [inline fieldservice_change_management/__manifest__.py]:
```suggestion
    'depends': [
        'fieldservice',
```

`fieldservice` already depends on `mail`.

File: fieldservice_change_management/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# Copyright (c) 2020 Pavlov Media <https://www.pavlovmedia.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    'name': 'Field Service - Change Management',
+    'summary': 'Manage Change Logs on Locations',
+    'author': 'Patrick Wilson: Pavlov Media',
```
Comment (max3903) [inline fieldservice_change_management/__manifest__.py]:
```suggestion
    'author': 'Pavlov Media, Odoo Community Association (OCA)',
```

File: fieldservice_change_management/report/change_log_reports.xml [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+                        <th class="text-left">Changed On</th>
+                        <th class="text-left">Changed By</th>
+                    </tr>
+                </thead>
+                <tbody>
+                  <tr>
+                    <td><span t-field="doc.name"/></td>
+                    <td class="text-left"><span t-field="doc.type_id"/></td>
+                    <td class="text-left"><span t-field="doc.description"/></td>
+                    <td class="text-left"><span t-field="doc.change_date"/></td>
```
Comment (brian10048) [inline fieldservice_change_management/report/change_log_reports.xml]: KeyError: 'change_date'

Error to render compiling AST
KeyError: 'change_date'
Template: 1449
Path: /templates/t/t/t/t/t/t/table/tbody/tr/td[4]/span
Node: <span t-field="doc.change_date"/>

```suggestion
                    <td class="text-left"><span t-field="doc.implemented_on"/></td>
```

File: fieldservice_change_management/report/change_log_reports.xml [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+                        <th class="text-left">Changed By</th>
+                    </tr>
+                </thead>
+                <tbody>
+                  <t t-foreach="doc.change_log_ids" t-as="log">
+                    <tr>
+                      <td><span t-field="log.name"/></td>
+                      <td class="text-left"><span t-field="log.type_id"/></td>
+                      <td class="text-left"><span t-field="log.description"/></td>
+                      <td class="text-left"><span t-field="log.change_date"/></td>
```
Comment (brian10048) [inline fieldservice_change_management/report/change_log_reports.xml]:
```suggestion
                      <td class="text-left"><span t-field="log.implemented_on"/></td>
```

**Review by brian10048 — CHANGES_REQUESTED** [review]
Need to fix key error on report for wrong field name.

In general this is good with me.

Additional thoughts:
I wonder if it should be split up with a standalone 'Change Log' app and another smaller module to implement field service change logs.  That way change logs could be used for other modules or records in Odoo.

**Comment by patrickrwilson** [comment]
@brian10048 Thanks for your review. I'm curious of what others feel about your suggestion to split this into a base module and then a separate field service module. I understand that that could be more useful for future expansion if needed. I just did it specific to field service because for our use that's all that's needed. 

If we were to split this, what repo do you think the base module should go in?

**Comment by brian10048** [comment]
@patrickrwilson Totally understood.  I'm fine with it being here for now.

@max3903 might have a better suggestion for a repo if you were to split the module

**Comment by max3903** [comment]
@patrickrwilson I would put it in https://github.com/OCA/server-tools

**Comment by patrickrwilson** [comment]
> @patrickrwilson I would put it in https://github.com/OCA/server-tools

@max3903 It seems this is more related to field service than server tools to me. Any way we can go ahead and leave it here?

**Comment by max3903** [comment]
@patrickrwilson Can you fix the Runbot warnings:

```
2020-05-27 15:52:08,442 181 WARNING openerp_test odoo.models: change.log.type.create() with unknown fields: owner_sequence
```

**Comment by patrickrwilson** [comment]
> Can you fix the Runbot warnings:
> 
> ```
> 2020-05-27 15:52:08,442 181 WARNING openerp_test odoo.models: change.log.type.create() with unknown fields: owner_sequence
> ```

@max3903 Fixed and green

**Review by osi-scampbell — APPROVED** [review]
@max3903 Tested and verified on Test server.

**Pull request merged**